sw/device/tests/sim_dv/ast_clk_rst_inputs.c

```diff
@@ -0,0 +1,577 @@
+ *  status is seen, likewise for fatal events.
+ *  Finally, check for correct capture of cause in alert handler.
+ */
+static void test_event(uint32_t idx, dif_toggle_t fatal, bool set_event) {
+  if (set_event) {
+    // Configure event fatality
+    CHECK_DIF_OK(dif_sensor_ctrl_set_alert_fatal(&sensor_ctrl, idx, fatal));
+
+    // Trigger event
+    CHECK_DIF_OK(dif_sensor_ctrl_set_ast_event_trigger(&sensor_ctrl, idx,
+                                                       kDifToggleEnabled));
+    // wait for events to set
+    IBEX_SPIN_FOR(get_events(fatal) > 0, 1);
+
+    // Check for the event in ast sensor_ctrl
+    // if the event is not set, error
+    CHECK(((get_events(fatal) >> idx) & 0x1) == 1,
+          "Event %d not observed in AST", idx);
+
+    // check the opposite fatality setting, should not be set
+    CHECK(((get_events(!fatal) >> idx) & 0x1) == 0,
+          "Event %d observed in AST when it should not be", idx);
+  } else {
+    // clear event trigger
+    clear_event(idx, fatal);
+
+    // check whether alert handler captured the event
+    check_alert_state(fatal);
+  }
+};
+
+void init_units() {
+  CHECK_DIF_OK(dif_pwrmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR), &pwrmgr));
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+  CHECK_DIF_OK(dif_entropy_src_init(
+      mmio_region_from_addr(TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR), &entropy_src));
+  CHECK_DIF_OK(dif_aon_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR), &aon_timer));
+  CHECK_DIF_OK(dif_rv_plic_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR), &rv_plic));
+  CHECK_DIF_OK(dif_sensor_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_SENSOR_CTRL_BASE_ADDR), &sensor_ctrl));
+  CHECK_DIF_OK(dif_alert_handler_init(
+      mmio_region_from_addr(TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR),
+      &alert_handler));
+  CHECK_DIF_OK(dif_rv_plic_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR), &plic));
+  CHECK_DIF_OK(dif_clkmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_CLKMGR_AON_BASE_ADDR), &clkmgr));
+  CHECK_DIF_OK(dif_adc_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_ADC_CTRL_AON_BASE_ADDR), &adc_ctrl));
+}
+
+/**
+ *  configure adc module
+ */
+static void configure_adc_ctrl(const dif_adc_ctrl_t *adc_ctrl) {
+  CHECK_DIF_OK(dif_adc_ctrl_set_enabled(adc_ctrl, kDifToggleDisabled));
+  CHECK_DIF_OK(dif_adc_ctrl_reset(adc_ctrl));
+  CHECK_DIF_OK(dif_adc_ctrl_configure(
+      adc_ctrl, (dif_adc_ctrl_config_t){
+                    .mode = kDifAdcCtrlLowPowerScanMode,
+                    .num_low_power_samples = kNumLowPowerSamples,
+                    .num_normal_power_samples = kNumNormalPowerSamples,
+                    .power_up_time_aon_cycles = kPowerUpTimeAonCycles,
+                    .wake_up_time_aon_cycles = kWakeUpTimeAonCycles}));
```

Comment:
@sha-ron @ariel9678 can you update this part and use the function suggested on https://github.com/lowRISC/opentitan/pull/16541#discussion_r1035306866 to calculate the AON cycles?

We can do this in a follow-up PR.

Reply:
yeah agree with @msfschaffner , doing a microsecnds to cycles conversion wil be more robust long term, since the mircosecond value can change, and we can maybe define that per top.

Reply:
Sure I can do it, just to understand do you mean to convert the  kPowerUpTimeAonCycles & kWakeUpTimeAonCycles values to microsecnds and than using the aon_timer_testutils_get_aon_cycles_from_us(uint64_t microseconds) function to convert it back to aon cycls?  
The kPowerUpTimeAonCycles & kWakeUpTimeAonCycles are values being written to [adc_pd_ctl](https://docs.opentitan.org/hw/ip/adc_ctrl/doc/#Reg_adc_pd_ctl) register. and not calculated values.

Reply:
Yeah exactly. We would like tests like this one to be portable, and since some platforms may have a different AON clock frequency (e.g. FPGA), we are using these conversion functions to make sure that the configuration is correct for a given platform. This is btwy similar to setting baudrates on UART, because the peripheral clock may be different between platforms.

Reply:
Done in #16729